app/controllers/verify/finance_controller.rb

```diff
@@ -7,13 +7,18 @@ class FinanceController < ApplicationController
     before_action :confirm_step_allowed
     before_action :submit_idv_form, only: [:create]
     before_action :submit_idv_job, only: [:create]
+    before_action :refresh_if_not_ready, only: [:show]
```

Comment:
Why not attach this to the existing "new" action? That way `create` can stay as it was and you wouldn't need to `redirect_to verify_finance_result_path`. Same with the other controllers.